### PR TITLE
DDCE-5030: Migrate Joda time to Java time : bc-passengers-frontend

### PR DESCRIPTION
**Pull request by robpetterson**

### CODE REVIEW

File: test/controllers/DeclarationRetrievalControllerSpec.scala [hunk 1]
```diff
@@ -301,7 +300,7 @@ class DeclarationRetrievalControllerSpec extends BaseSpec {
         "Newcastle Airport",
         "",
         LocalDate.parse("2021-04-01"),
```
Comment (frikit) [inline test/controllers/DeclarationRetrievalControllerSpec.scala:302]: use your new method parseLocalDate

File: test/controllers/ZeroDeclarationControllerSpec.scala [hunk 2]
```diff
@@ -199,7 +198,7 @@ class ZeroDeclarationControllerSpec extends BaseSpec {
     "LHR",
     "",
     LocalDate.parse("2018-11-12"),
```
Comment (frikit) [inline test/controllers/ZeroDeclarationControllerSpec.scala:200]: use your new method parseLocalDate

File: test/services/DateTimeProviderServiceSpec.scala [hunk 3]
```diff
@@ -14,21 +14,3 @@
  * limitations under the License.
  */
 
-package services
-
-import util.BaseSpec
-
-import java.time.LocalDate
-
-class DateTimeProviderServiceSpec extends BaseSpec {
```
Comment (frikit) [inline test/services/DateTimeProviderServiceSpec.scala:23]: why you deleted these tests? I can see main class is still there anad not have a unit tests in place

File: test/services/DeclarationServiceSpec.scala [hunk 4]
```diff
@@ -82,7 +81,7 @@ class DeclarationServiceSpec extends BaseSpec with ScalaFutures {
     "LHR",
     "",
     LocalDate.parse("2018-05-31"),
```
Comment (frikit) [inline test/services/DeclarationServiceSpec.scala:83]: use your new method parseLocalDate

File: test/services/DeclarationServiceSpec.scala [hunk 5]
```diff
@@ -2392,7 +2415,7 @@ class DeclarationServiceSpec extends BaseSpec with ScalaFutures {
         "LHR",
         "",
         LocalDate.parse("2018-05-31"),
```
Comment (frikit) [inline test/services/DeclarationServiceSpec.scala:2417]: use your new method parseLocalDate

File: test/services/PayApiServiceSpec.scala [hunk 6]
```diff
@@ -186,7 +188,7 @@ class PayApiServiceSpec extends BaseSpec {
       "LHR",
       "",
       LocalDate.parse("2018-11-12"),
```
Comment (frikit) [inline test/services/PayApiServiceSpec.scala:190]: use your new method parseLocalDate

File: test/services/PayApiServiceSpec.scala [hunk 7]
```diff
@@ -414,8 +416,8 @@ class PayApiServiceSpec extends BaseSpec {
       val uiWithBstArrival: UserInformation = userInformation.copy(
         selectPlaceOfArrival = "",
         enterPlaceOfArrival = "LHR",
-        dateOfArrival = LocalDate.parse("2018-7-12"),
-        timeOfArrival = LocalTime.parse("12:20 pm", DateTimeFormat.forPattern("hh:mm aa"))
+        dateOfArrival = LocalDate.parse("2018-07-12"),
+        timeOfArrival = LocalTime.parse("12:20 pm", DateTimeFormatter.ofPattern("hh:mm a",Locale.UK))
```
Comment (frikit) [inline test/services/PayApiServiceSpec.scala:420]: use your new method parseLocalTime

File: test/services/PayApiServiceSpec.scala [hunk 8]
```diff
@@ -414,8 +416,8 @@ class PayApiServiceSpec extends BaseSpec {
       val uiWithBstArrival: UserInformation = userInformation.copy(
         selectPlaceOfArrival = "",
         enterPlaceOfArrival = "LHR",
-        dateOfArrival = LocalDate.parse("2018-7-12"),
-        timeOfArrival = LocalTime.parse("12:20 pm", DateTimeFormat.forPattern("hh:mm aa"))
+        dateOfArrival = LocalDate.parse("2018-07-12"),
```
Comment (frikit) [inline test/services/PayApiServiceSpec.scala:419]: this should work as 7 if I understand pattern corrctly please revert from 07 to 7 and
use your new method parseLocalDate

File: test/services/PayApiServiceSpec.scala [hunk 9]
```diff
@@ -445,8 +447,8 @@ class PayApiServiceSpec extends BaseSpec {
...
```
Comment (frikit) [inline test/services/PayApiServiceSpec.scala:450]: this should work as 7 if I understand pattern corrctly please revert from 07 to 7 and
use your new method parseLocalDate

File: test/services/PayApiServiceSpec.scala [hunk 10]
```diff
@@ -445,8 +447,8 @@ class PayApiServiceSpec extends BaseSpec {
...
+        timeOfArrival = LocalTime.parse("12:20 pm", DateTimeFormatter.ofPattern("hh:mm a",Locale.UK))
```
Comment (frikit) [inline test/services/PayApiServiceSpec.scala:451]: use your new method parseLocalTime

File: test/services/PayApiServiceSpec.scala [hunk 11]
```diff
@@ -485,8 +487,8 @@ class PayApiServiceSpec extends BaseSpec {
...
```
Comment (frikit) [inline test/services/PayApiServiceSpec.scala:491]: use your new method parseLocalTime

File: test/services/PayApiServiceSpec.scala [hunk 12]
```diff
@@ -485,8 +487,8 @@ class PayApiServiceSpec extends BaseSpec {
       val uiWithBstArrival: UserInformation = userInformation.copy(
         selectPlaceOfArrival = "",
         enterPlaceOfArrival = "LHR",
-        dateOfArrival = LocalDate.parse("2018-7-12"),
-        timeOfArrival = LocalTime.parse("12:20 pm", DateTimeFormat.forPattern("hh:mm aa"))
+        dateOfArrival = LocalDate.parse("2018-07-12"),
```
Comment (frikit) [inline test/services/PayApiServiceSpec.scala:490]: this should work as 7 if I understand pattern corrctly please revert from 07 to 7 and
use your new method parseLocalDate

File: test/services/UserInformationServiceSpec.scala [hunk 13]
```diff
@@ -72,7 +71,7 @@ class UserInformationServiceSpec extends BaseSpec {
             "Newcastle Airport",
             "",
             LocalDate.parse("2018-08-31"),
```
Comment (frikit) [inline test/services/UserInformationServiceSpec.scala:73]: use your new method parseLocalDate

File: test/services/UserInformationServiceSpec.scala [hunk 14]
```diff
@@ -90,7 +89,7 @@ class UserInformationServiceSpec extends BaseSpec {
                 "Newcastle Airport",
                 "",
                 LocalDate.parse("2018-08-31"),
```
Comment (frikit) [inline test/services/UserInformationServiceSpec.scala:91]: use your new method parseLocalDate

File: test/views/declaration/ZeroDeclarationViewSpec.scala [hunk 15]
```diff
@@ -108,7 +109,7 @@ class ZeroDeclarationViewSpec extends BaseViewSpec {
     selectPlaceOfArrival = "",
     enterPlaceOfArrival = "Newcastle Airport",
     dateOfArrival = LocalDate.parse("2023-05-06"),
```
Comment (frikit) [inline test/views/declaration/ZeroDeclarationViewSpec.scala:111]: use your new method parseLocalDate

**Review by frikit — CHANGES_REQUESTED** [review]

File: a11y/AccessibilityConstants.scala [hunk 16]
```diff
@@ -130,7 +132,7 @@ trait AccessibilityConstants {
     selectPlaceOfArrival = "",
     enterPlaceOfArrival = "Newcastle Airport",
     dateOfArrival = LocalDate.parse("2023-05-06"),
```
[the remark this answers is not in the document]
Comment (robpetterson) [reply a11y/AccessibilityConstants.scala:134]: Changed

File: test/controllers/DeclarationRetrievalControllerSpec.scala (hunk 1 above)
Comment (robpetterson) [reply test/controllers/DeclarationRetrievalControllerSpec.scala:302]: Changed

File: test/controllers/ZeroDeclarationControllerSpec.scala (hunk 2 above)
Comment (robpetterson) [reply test/controllers/ZeroDeclarationControllerSpec.scala:200]: Changed

File: test/services/DateTimeProviderServiceSpec.scala (hunk 3 above)
Comment (robpetterson) [reply test/services/DateTimeProviderServiceSpec.scala:23]: Changed

File: test/services/PayApiServiceSpec.scala (hunk 12 above)
Comment (robpetterson) [reply test/services/PayApiServiceSpec.scala:490]: Changed

File: test/services/UserInformationServiceSpec.scala (hunk 13 above)
Comment (robpetterson) [reply test/services/UserInformationServiceSpec.scala:73]: Changed

File: test/services/UserInformationServiceSpec.scala (hunk 14 above)
Comment (robpetterson) [reply test/services/UserInformationServiceSpec.scala:91]: Changed

File: test/views/declaration/ZeroDeclarationViewSpec.scala (hunk 15 above)
Comment (robpetterson) [reply test/views/declaration/ZeroDeclarationViewSpec.scala:111]: Changed

**Comment by robpetterson** [comment]
Replaced LocalDate.parse and LocalTime.parse to parseLocalDate and parseLocalTime throughout files

**Review by frikit — APPROVED** [review]

**Pull request merged**